backend/httpclient/tracing_middleware.go

```diff
@@ -46,10 +43,13 @@ func TracingMiddleware(tracer trace.Tracer) Middleware {
 			otel.GetTextMapPropagator().Inject(ctx, propagation.HeaderCarrier(req.Header))
 			res, err := next.RoundTrip(req)
 
-			span.SetAttributes(attribute.String("http.url", req.URL.String()))
-			span.SetAttributes(attribute.String("http.method", req.Method))
+			span.SetAttributes(
+				semconv.HTTPURL(req.URL.String()),
+				semconv.HTTPMethod(req.Method),
+			)
```

Comment:
The following have all been deprecated:

- `http.url`
- `http.method`
- `http.status_code`

and `http.content_length` is not standard

https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/#deprecated-http-attributes

This is the same in Grafana core:
https://github.com/grafana/grafana/blob/a4bb4c84002374f24cfc44efa42304e131353d89/pkg/infra/httpclient/httpclientprovider/tracing_middleware.go

but shall we maybe also add the new non-deprecated/standard ones alongside those (both in core and sdk)?

Reply:
Yes I noticed and was considering to change this and use the https://github.com/open-telemetry/opentelemetry-go-contrib/blob/c44d632df195200dee20d76b9b486d0349bfa9ea/instrumentation/net/http/otelhttp/transport.go#L51 to standardize on things. However, I think the first step is to make Grafana reuse the SDK tracing middleware why I suggest to keep using the deprecated fields for now.